dynamo-adapter.js

```diff
@@ -0,0 +1,110 @@
+const {promisify} = require('util');
+const {DynamoTable} = require('./slim');
+
+const _isCallback = (fn) => typeof fn === 'function';
+
+const _validateCall = (methodName, args, minLength) => {
+  if (args.length < minLength)
+    throw new Error(`Incorrect args supplied to method ${methodName}: expected: ${minLength} got: ${args.length}`);
+}
+
+/**
+  Provides an adapter for the DynamoTable class to allow for promisified
+  api usage when omitting the cb parameter. Does not use ES6 classes
+  because some dependents use assign to copy the instance which is only a
+  shallow copy and insufficient to clonean ES6 class instance.
+*/
+function DynamoAdapter(name, options) {
```

Comment:
hmm es6 has classes, so you can actually do
```javascript
class DynamoAdapter {
  constructor(name, options) {
    ...
  }
  ...
}
```

Reply:
Actually I wrote the whole thing in es6 but in some place where its used its 'copied' with `_.omit` which actually does not include properties of es6 classes.

There is actually a difference to how these properties are taken from an instance when using this form vs es6 class form. I changed the whole thing back to this as it meant I didnt need to change the consuming logic which still does the `_.omit`. Later I did have to make a couple more changes around that location but I think we can just keep this style for now then I wont have to change that whole piece with the average cloning attempt.